src/data/types/recordid.ts

```diff
@@ -51,33 +51,37 @@ export class RecordId<Tb extends string = string> extends Value {
  * A SurrealQL string-represented record ID value.
  */
 export class StringRecordId extends Value {
-	public readonly rid: string;
-
-	constructor(rid: string | StringRecordId | RecordId) {
-		super();
-
-		// In some cases the same method may be used with different data sources
-		// this can cause this method to be called with an already instanced class object.
-		if (rid instanceof StringRecordId) {
-			this.rid = rid.rid;
-		} else if (rid instanceof RecordId) {
-			this.rid = rid.toString();
-		} else if (typeof rid === "string") {
-			this.rid = rid;
-		} else {
-			throw new SurrealDbError("String Record ID must be a string");
-		}
-	}
+    public readonly rid: string;
+
+    constructor(rid: string | StringRecordId | RecordId) {
+        super();
+
+        if (rid instanceof StringRecordId) {
+            this.rid = rid.rid;
+        } else if (rid instanceof RecordId) {
+            this.rid = rid.toString();
+        } else if (typeof rid === "string") {
+            // Handle long numeric string IDs in the constructor
+            const [table, id] = rid.split(':');
+            if (id && /^\d{15,}$/.test(id)) {
+                this.rid = `${table}:⟨${id}⟩`;
+            } else {
+                this.rid = rid;
+            }
+        } else {
+            throw new SurrealDbError("String Record ID must be a string");
+        }
+    }
```

Comment:
We assume that the user passes a correct record id here. We purposefully do not parse record ids in the JS SDK, as they can contain a huge subset of surrealql values, making them vastly complex to parse outside the surrealql parser. Could you revert this part of your PR?